### PR TITLE
Correctly purge the cache when an app is disabled via cli

### LINKED ISSUE #15359: APC isn't properly cleanup after enabling / disabling apps from the command-line
### Steps to reproduce
1. Make sure you have installed APC in ownCloud (checked with APCu installed php5-apc package in ubuntu 14.04)
2. Go to the settings apps (enabled section) and then change to the disabled section
3. From the command-line enable a disabled app
4. Go to the settings apps - enabled section

### Expected behaviour
App should appear as enabled

### Actual behaviour
App still shows as disabled

### Server configuration
**Operating system**: ubuntu 14.04

**Web server:** apache 2.4.7

**Database:** mysql

**PHP version:** 5.5.4

**ownCloud version:** OC 8.0.3RC1

**Updated from an older ownCloud or fresh install:** fresh install

### APC configuration (from phpinfo page)
```
APC support 	Emulated

apcu
APCu Support	enabled
Version 	4.0.2
APCu Debugging 	Disabled
MMAP Support 	Enabled
MMAP File Mask 	no value
Serialization Support 	php, eval
Revision 	$Revision: 328290 $
Build Date 	Feb 19 2014 09:18:08

Directive	Local Value	Master Value
apc.coredump_unmap	Off	Off
apc.enable_cli	On	On
apc.enabled	On	On
apc.entries_hint	4096	4096
apc.gc_ttl	3600	3600
apc.mmap_file_mask	no value	no value
apc.preload_path	no value	no value
apc.rfc1867	Off	Off
apc.rfc1867_freq	0	0
apc.rfc1867_name	APC_UPLOAD_PROGRESS	APC_UPLOAD_PROGRESS
apc.rfc1867_prefix	upload_	upload_
apc.rfc1867_ttl	3600	3600
apc.serializer	php	php
apc.shm_segments	1	1
apc.shm_size	32M	32M
apc.slam_defense	On	On
apc.smart	0	0
apc.ttl	0	0
apc.use_request_time	On	On
apc.writable	/tmp	/tmp
```
You'll need to wait until the cache expires

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #15359

@jvillafanez @LukasReschke @MorrisJobke

### CODE REVIEW

**Comment by karlitschek** [comment]
:+1:

**Comment by nickvergessen** [comment]
@owncloud-bot retest this please

File: lib/private/app/appmanager.php [hunk 1]
```diff
@@ -54,10 +58,11 @@ class AppManager implements IAppManager {
 	 * @param \OCP\IAppConfig $appConfig
 	 * @param \OCP\IGroupManager $groupManager
 	 */
-	public function __construct(IUserSession $userSession, IAppConfig $appConfig, IGroupManager $groupManager) {
+	public function __construct(IUserSession $userSession, IAppConfig $appConfig, IGroupManager $groupManager, ICacheFactory $memCacheFactory) {
```
Comment (LukasReschke) [inline lib/private/app/appmanager.php]: PHPDoc needs to get adjusted as well then. And while at it maybe add these entries on a new line?

**Comment by LukasReschke** [comment]
Besides my nitpicks: +1

File: lib/private/app/appmanager.php (hunk 1 above)
Comment (nickvergessen) [inline lib/private/app/appmanager.php]: fixed

**Comment by jnfrmarks** [comment]
Backport to 8.0.3?

**Comment by karlitschek** [comment]
please backport

**Pull request merged**